src/extension.ts

```diff
@@ -25,24 +25,53 @@ import * as is from 'vscode-languageclient/lib/utils/is';
 
 const CONFIGURATION = RLSConfiguration.loadFromWorkspace();
 
+function getSysroot(env: Object): string | Error {
+    const rustcSysroot = child_process.spawnSync('rustc', ['--print', 'sysroot'], { env });
+
+    if (rustcSysroot.error) {
+        return new Error(`Error running \`rustc\`: ${rustcSysroot.error}`);
+    }
+
+    if (rustcSysroot.status > 0) {
+        return new Error(`Error getting sysroot from \`rustc\`: exited with \`${rustcSysroot.status}\``);
+    }
+    
+    if (!rustcSysroot.stdout || typeof rustcSysroot.stdout.toString !== 'function') {
+        return new Error(`Couldn't get sysroot from \`rustc\`: Got no ouput`);
+    }
+
+    const sysroot = rustcSysroot.stdout.toString()
+        .replace('\n', '').replace('\r', '');
+
+    return sysroot;
+}
+
 // Make an evironment to run the RLS.
 // Tries to synthesise RUST_SRC_PATH for Racer, if one is not already set.
 function makeRlsEnv(): any {
     const env = process.env;
+
     if (process.env.RUST_SRC_PATH) {
         return env;
-    } else {
-        // rustc --print sysroot + lib/rustlib/src/rust/src
-        const result = child_process.spawnSync('rustc', ['--print', 'sysroot']);
-        if (result.status > 0) {
-            console.log('error running rustc for sysroot');
-            return env;
+    }
+
+    let result = getSysroot(env);
+    if (result instanceof Error) {
+        console.info(result.message);
+        console.info(`Let's retry with extended $PATH`);
+        env.PATH = `${env.HOME || '~'}/.cargo/bin:${env.PATH || ''}`;
+        result = getSysroot(env);
+
+        if (result instanceof Error) {
+            console.warn(result);
```

Comment:
Could you use `window.showWarningMessage` here so the user will see it if they don't have their console open. Also, could you make a user-friendly message about not being able to set RUST_SRC_PATH because of an error reading the Rust sysroot?

Reply:
Sure, will push in a second. I didn't want to introduce a new user-visible warning, but f you want one, I'll be happy to make one :)